### PR TITLE
Enable in-game keyboard to optionally activate key bindings

### PR DESCRIPTION
**Pull request by Techjar**
Disabled by default, since I think it could be bad UX if someone accidentally activates bindings they don't intend to. We can always change it if we determine otherwise. If disabled, a message will pop up in chat when you type without a screen open.

I also included a small refactor of VRSettings for code clarity and some additional functionality.

### CODE REVIEW

**Comment by soup587** [comment]
hell yeah, been waiting a while for this

**Comment by fayer3** [comment]
pressKeyHelper should probably be moved to the InputSimulator, don't need to have the exact same code in two classes

**Comment by fayer3** [comment]
and could it be changed to a separate press and release, for stuff that wants bindings pressesd for a longer time, like creates pondering

**Comment by fayer3** [comment]
is it okay, if I push to your repo, or do you want to do that yourself

**Comment by fayer3** [comment]
but it works fine from what I could see, maybe an option to not close the keyboard, when a new screen pops up would be nice, but that can be done later

**Comment by fayer3** [comment]
> and could it be changed to a separate press and release, for stuff that wants bindings pressesd for a longer time, like creates pondering

easy to to for the physical keyboard, a bit more work for the pointer one

File: common/src/main/java/org/vivecraft/client_vr/gui/GuiKeyboard.java [hunk 1]
```diff
@@ -42,8 +44,10 @@ public void init() {
                 }
 
                 String s2 = String.valueOf(c0);
+                final int code = k < this.dataholder.vrSettings.keyboardCodes.length ? this.dataholder.vrSettings.keyboardCodes[k] : GLFW.GLFW_KEY_UNKNOWN;
```
Comment (fayer3) [inline common/src/main/java/org/vivecraft/client_vr/gui/GuiKeyboard.java]: k here should be l1, maybe would be a good idea to readd the named variables here too

**Comment by Techjar** [comment]
I can take care of it myself later, it's no problem.

**Comment by Techjar** [comment]
> and could it be changed to a separate press and release, for stuff that wants bindings pressesd for a longer time, like creates pondering

This is difficult to do as there's a lot of cases where the key could become "stuck" in the pressed state.

**Comment by Techjar** [comment]
Changes are done, though I didn't actually test them.

**Pull request merged**